src/cinder/audio/cocoa/DeviceManagerAudioSession.mm

```diff
@@ -264,18 +264,21 @@ @implementation AudioSessionNotificationHandlerImpl
 
 - (void)handleInterruption:(NSNotification *)notification
 {
-	NSUInteger interruptionType = [[notification.userInfo valueForKey:AVAudioSessionInterruptionTypeKey] unsignedIntValue];
+	NSUInteger interruptionType = [[notification.userInfo objectForKey:AVAudioSessionInterruptionTypeKey] unsignedIntValue];
 
 	if( interruptionType == AVAudioSessionInterruptionTypeBegan ) {
 		mDeviceManager->beginInterruption();
 	}
 	else if( interruptionType == AVAudioSessionInterruptionTypeEnded ) {
-		NSUInteger interruptionOption = [[notification.userInfo valueForKey:AVAudioSessionInterruptionOptionKey] unsignedIntValue];
-		if( interruptionOption == AVAudioSessionInterruptionOptionShouldResume ) {
-			mDeviceManager->endInterruption(true);
-		} else {
-			mDeviceManager->endInterruption(false);
-		}
+		UIApplication *app = [UIApplication sharedApplication];
+		AVAudioSession *session = [AVAudioSession sharedInstance];
+
+		auto resumeImmediately = app.applicationState == UIApplicationStateActive ||
```

Comment:
I was imagining that DeviceManagerAudioSession kept an `mWasInterrupted` bool, and then in the case of an ambient session category, if this is true during its app activated signal callback then it would call the interruption ended signal (in the regular case of a play-and-record session category, this flag would have already been set back to false as the interruption ended callback was immediately called when the obj-c notification fired).

PM me if that is confusing, this is super close so happy to hash it out and get it into master branch.